lib/spack/spack/build_systems/cuda.py

```diff
@@ -92,9 +92,9 @@ def cuda_flags(arch_list):
     conflicts('%gcc@7:', when='+cuda ^cuda@:9.1 target=x86_64:')
     conflicts('%gcc@8:', when='+cuda ^cuda@:10.0.130 target=x86_64:')
     conflicts('%gcc@9:', when='+cuda ^cuda@:10.2.89 target=x86_64:')
-    conflicts('%gcc@:4', when='+cuda ^cuda@11.0.2: target=x86_64:')
-    conflicts('%gcc@10:', when='+cuda ^cuda@:11.0.3 target=x86_64:')
-    conflicts('%gcc@11:', when='+cuda ^cuda@:11.1.0 target=x86_64:')
+    conflicts('%gcc@:4', when='+cuda ^cuda@11.0.0: target=x86_64:')
+    conflicts('%gcc@:5', when='+cuda ^cuda@11.4.0: target=x86_64:')
+    conflicts('%gcc@10:', when='+cuda ^cuda@11.0.0: target=x86_64:')
```

Comment:
This would also mean that newer versions of cuda would conflict with newer gcc versions, so it's better to have an upper bound for gcc for a lower bound of cuda or the other way around.

We should just check better that whenever a new cuda minor version is released we actually bump the upperbound for cuda on the conflict rule.